test/integration/node-specific/db.test.js

```diff
@@ -1,37 +1,38 @@
 'use strict';
 
+import { MongoServerError } from 'mongodb-legacy';
+
 const { setupDatabase, assert: test } = require(`../shared`);
 const { expect } = require('chai');
-const { Db, MongoClient } = require('../../mongodb');
+const { MongoClient } = require('../../mongodb');
 
 describe('Db', function () {
   before(function () {
     return setupDatabase(this.configuration);
   });
 
-  it('shouldCorrectlyHandleIllegalDbNames', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded'] }
-    },
-
-    test: done => {
-      const client = { bsonOptions: {} };
-      expect(() => new Db(client, 5)).to.throw('Database name must be a string');
-      expect(() => new Db(client, '')).to.throw('Database name cannot be the empty string');
-      expect(() => new Db(client, 'te$t')).to.throw(
-        "database names cannot contain the character '$'"
-      );
-      expect(() => new Db(client, '.test', function () {})).to.throw(
-        "database names cannot contain the character '.'"
-      );
-      expect(() => new Db(client, '\\test', function () {})).to.throw(
-        "database names cannot contain the character '\\'"
-      );
-      expect(() => new Db(client, 'test test', function () {})).to.throw(
-        "database names cannot contain the character ' '"
-      );
-      done();
-    }
+  context('when given illegal db name', function () {
+    let client;
+    let db;
+
+    beforeEach(function () {
+      client = this.configuration.newClient();
+    });
+
+    afterEach(async function () {
+      db = undefined;
+      await client.close();
+    });
+
+    it('should throw error on server only', async function () {
+      try {
+        db = client.db('a\x00b');
+        await db.createCollection('spider');
+        expect.fail('a MongoServerError was expected due to illegal db name');
+      } catch (error) {
+        expect(error instanceof MongoServerError);
```

Comment:
Let's use the chai `expect(error).to.be.instanceOf()` assertion here to get a more detailed error message

Reply:
Sounds good, as it turns out its not an instance of MongoServerError, but rather MongoServerError is the name of the error, so I added a check for that instead.